Add How-to SLSA for organizations and How-to SLSA for implementers (#823)
slsa-framework/slsa: docs/how-to-orgs.md

---
title: How to SLSA for organizations
description: If you're looking to jump straight in and try SLSA, here's a quick start guide for the steps to take to reach the different SLSA levels.
layout: standard
---

This is a quick-start guide for organizations that want to adopt SLSA. Your

> organization has two major responsibilities: choosing a SLSA level for your

Reviewer: choosing a target SLSA level?

Author: Done

organization, and selecting tools that support your desired SLSA level.

## Choosing your SLSA level

For all [SLSA levels](/spec/v1.0/levels.md), you follow the same steps:

1)  Generate provenance, i.e., document your build process
2)  Make the provenance available, to allow downstream users to verify it

What differs for each level is the robustness of the build and provenance. For
more information about SLSA provenance see
[https://slsa.dev/provenance/](/provenance/v1.md).

SLSA Build levels are progressive: SLSA Build L3 includes all the guarantees of
SLSA Build L2, and SLSA Build L2 includes all the guarantees of SLSA Build L1.
Currently, though, the work required to achieve lower SLSA levels will not
necessarily accrue toward the work needed for higher levels, because achieving a
higher level may require migrating to a different build platform altogether.

> For that reason, **you should start with the highest level that’s possible for
> your project or organization to avoid wasted work**.

Reviewer 2: This is a nice call out to help folks avoid wasted work, but I'd phrase this differently. We have always suggested that SLSA is incrementally attainable, that folks can start at one level and build up and needn't dive straight into the highest level. How about:

```suggestion
For that reason, you should set a target level for your project or
organization to work towards and choose a build platform which supports the
target level, so as to avoid wasted work.
```

Reviewer: I see both sides here --- the messaging has been to gradually move up the ladder, but in the current state, does work you put toward L1 accrue toward L2, or would you need to start over as an org? I know that in the past there has been the misunderstanding that you HAVE to start at L1 and tick off the boxes there before moving on to L2. The suggested wording might support that interpretation.

Reviewer 3: I had a similar reaction to @joshuagl's when this wording came up as part of the update of the Get started page. Maybe expanding on the explanation along the lines of what you are describing here @olivekl would help.

<a id="tooling"></a>

## Choose appropriate tools

Your organization's tooling needs depend on what role your organization plays in
the software supply chain. Organizations can be software producers, software
consumers, or both. In practice, most organizations that produce software also
consume software.

### For software consumers

As a software consumer, you need to verify that the software you consume meets
your chosen SLSA Build level.

Ideally, [package ecosystems](/spec/v1.0/terminology.md#package-model) verify
SLSA provenance for the packages they distribute. Check with the package
ecosystem where you get software to see if they verify SLSA provenance. If they
do, then inspect its verification practices to ensure that they meet your
needs. If they do not, then you may want to request that they add SLSA support.

Regardless of whether your package ecosystem verifies SLSA provenance, you may
wish to verify SLSA provenance yourself using tools such as
[`slsa-verifier`](https://github.com/slsa-framework/slsa-verifier).

To learn more about the verification process, see
[Verifying Artifacts](/spec/v1.0/verifying-artifacts.md).

### For software producers

> As a software producer, you need to produce and distribute attestations that

Reviewer: Is it worth stating that you also have to follow the build level expectations, and linking to them?

Reviewer 3: yes, I think so. Otherwise it reads as if you can be conformant by merely producing attestations and independently of whether your build platform fulfills the requirements.

Author: Done

demonstrate your software meets your desired SLSA Build level. SLSA

> recommends either the [SLSA provenance format](/provenance/) or a
> [SLSA Verification Summary Attestation (VSA)](/attestation_summary/), although
> the SLSA specification allows for other formats.

Reviewer 2: is it worth adding a brief comment on when you would use provenance vs a VSA?
```suggestion
recommends either the [SLSA provenance format](/provenance/) or,
when you wish to keep the details of your build pipeline confidential, a
[SLSA Verification Summary Attestation (VSA)](/attestation_summary/) although
the SLSA specification allows for other formats.
```

Reviewer 3: Is wanting to keep the details of your build platform the only reason you would use VSA? That's not my recollection but I have to admit not to have thought much about VSA so apologies if my comment is off.

Author: I'm not aware of any other reasons to use a VSA. I'll accept @joshuagl's suggestion

> Ideally, build automation tools produce SLSA provenance rather than individual

Reviewer 2: This introduces a new term, "build automation tools". Can we re-phrase these to use "build platform"?

Author: Done

developers. If your organization...

-   does not use a build automation tool, then consider adopting one that supports

> SLSA.

Reviewer: link to tooling suggestions here? Would they be the same as on the Dev How To page?

Reviewer 3: Eventually we should have a conformance program with a page of tools we can point to. For now I think we should just leave it as is.

Author: I'm going to leave this without a link for now.

-   uses a third-party build automation tool, then check with its provider to see
if they support generating SLSA provenance at your desired level. If they do,
then follow their instructions for producing provenance. If they do not, then
consider requesting that they add SLSA support or adopting a tool that does
support SLSA.
-   maintains its own build automation tool, then add support for generating SLSA
provenance.

For more information about producing provenance, see
[Producing artifact](/spec/v1.0/requirements) and
[Verifying build platforms](/spec/v1.0/verifying-systems).

Ideally, [package ecosystems](/spec/v1.0/terminology.md#package-model)
distribute provenance alongside packages. If your organization...

> -   distributes software through a third-party software ecosystem, then check with

Reviewer 2: ```suggestion
-   distributes software through a third-party package ecosystem, then check with
```

its provider to see if they support distributing SLSA provenance. If they do,
then follow their instructions for distributing provenance. If they do not, then
consider requesting that they add SLSA support or using a package ecosystem
that does support SLSA.
-   distributes software itself (e.g. directly to consumers), then add SLSA

> provenance to your software distributions.

Reviewer 2: ```suggestion
provenance to your package ecosystem.
```
Per our terminology and package model this is still a package ecosystem.

Reviewer 4: I don't think this is right, this is about sidestepping the package ecosystem entirely. A better replacement for this would probably be "package artifacts", "set of package artifacts" or "build outputs".

Author: Accepting @di's suggestion.

---
title: How to SLSA for organizations
description: If you're looking to jump straight in and try SLSA, here's a quick start guide for the steps to take to reach the different SLSA levels.
layout: standard
---

This is a quick-start guide for organizations that want to adopt SLSA. Your
organization has two major responsibilities: choosing a target SLSA level for
your organization, and selecting tools that support your desired SLSA level.

## Choosing your SLSA level

For all [SLSA levels](/spec/v1.0/levels.md), you follow the same steps:

1)  Generate provenance, i.e., document your build process
2)  Make the provenance available, to allow downstream users to verify it

What differs for each level is the robustness of the build and provenance. For
more information about SLSA provenance see
[https://slsa.dev/provenance/](/provenance/v1.md).

SLSA Build levels are progressive: SLSA Build L3 includes all the guarantees of

> SLSA Build L2, and SLSA Build L2 includes all the guarantees of SLSA Build L1.

Reviewer: I think this should state outright that you do not need to fulfill lower levels before moving on to the next level. Something like:

"...includes all the guarantees of SLSA Build L1. This means you can start at any level; the lower levels are not prerequisites for higher levels.

Furthermore, the work required to achieve the lower SLSA levels will not...."

Or feel free to phrase it differently. Just want to be very clear that these are not steps that need to be taken in order.
